includes/Core/Admin_Bar/Admin_Bar.php

```diff
@@ -149,6 +149,13 @@ public function is_active() {
 			return false;
 		}
 
+		// Always show admin bar on front page if user has capability.
+		// We need to shortcut this here, since this method run in 2 different hooks. In *_enqueue_scripts and admin_bar_menu.
+		// Since admin_bar_menu is called very late, the get_post below return different post object.
+		if ( is_front_page() && current_user_can( Permissions::VIEW_DASHBOARD ) ) {
```

Comment:
That shouldn't be here. The front page can be a post as well, so it should be covered below.